contracts/Market.sol

```diff
@@ -1336,7 +1374,9 @@ contract Market is
       isTransferSuccessful = _purchasingToken.transferFrom({
         from: from,
         to: suppliers[i],
-        amount: unrestrictedSupplierFee
+        amount: convertRemovalAmountToPurchasingTokenAmount(
+          unrestrictedSupplierFee
+        )
```

Comment:
same thoughts on perhaps doing this conversion earlier so the naming is less confusing